packages/url-loader/src/plugins/underlays.ts

```diff
@@ -140,56 +122,30 @@ export const underlaysPlugin = {
       // Positioning
 
       if (hasPosition) {
-        (Object.keys(position) as Array<keyof typeof position>).forEach(
-          (key) => {
-            if (!objectHasKey(qualifiersPosition, key)) return;
-            const { qualifier } = qualifiersPosition[key as string];
-            applied.push(`${qualifier}_${position[key]}`);
-          }
-        );
+        entriesOf(position).forEach(([key, value]) => {
+          if (!objectHasKey(qualifiersPosition, key)) return;
+          const { qualifier } = qualifiersPosition[key]!;
+          applied.push(`${qualifier}_${value}`);
+        });
       }
 
       // Layer Flags
       // Add flags to the primary layer transformation segment
       // @TODO: accept flag value
 
-      const activeLayerFlags = Array.isArray(layerFlags)
-        ? layerFlags
-        : [layerFlags];
-
-      activeLayerFlags.forEach((flag) => {
-        const { success } = flagsEnum.safeParse(flag);
+      const activeLayerFlags = isArray(layerFlags) ? layerFlags : [layerFlags];
 
-        if (!success) {
-          if (process.env.NODE_ENV === "development") {
-            console.warn(`Invalid flag ${flag}, not applying.`);
-          }
-          return;
-        }
-
-        primary.push(`fl_${flag}`);
-      });
+      activeLayerFlags.forEach((flag) => primary.push(`fl_${flag}`));
 
       // Applied Flags
       // Add flags to the fl_layer_apply transformation segment
       // @TODO: accept flag value
 
-      const activeAppliedFlags = Array.isArray(appliedFlags)
+      const activeAppliedFlags = isArray(appliedFlags)
         ? appliedFlags
         : [appliedFlags];
 
-      activeAppliedFlags.forEach((flag) => {
-        const { success } = flagsEnum.safeParse(flag);
-
-        if (!success) {
-          if (process.env.NODE_ENV === "development") {
-            console.warn(`Invalid flag ${flag}, not applying.`);
-          }
-          return;
-        }
```

Comment:
question: shouldn't we maintain this functionality? I think it is missing from a new approach

Reply:
I removed this based on feedback from @colbyfayock that we should be able to trust the input types to the functions themselves rather than performing runtime validation, which I think make sense if these flags are being invoked from an API directly.

If that's not the case, it would be easy enough to refactor those union definitions to tuples and then extract them back out to see if they're in the list, but in general I think the reasoning is good that if you have a typed API, you should need to revalidate all your inputs.

Reply:
i dont feel particularly strongly one way or another here. i always thought that providing more console.logs or throws that move errors closer to the developer could be helpful but not at the expense of trying to maintain something like zod, so moreso if there is good opportunity i think we should leverage it but not for any kind of significant overhead. this example was using zod, so it would need refactoring

@Baroshem any thoughts around that?